theseus_gui/src/routes/settings/accounts.svelte

```diff
@@ -0,0 +1,97 @@
+<script>
+  import { Button } from 'omorphia';
+  import { invoke } from '@tauri-apps/api/tauri';
+  import { onMount } from 'svelte';
+
+  let user = { uninitialized: true };
+
+  function fetchUser() {
+    invoke('auth_users')
+      .then((res) => {
+        user = res === [] ? { uninitialized: false } : res[0];
+      })
+      .catch((error) => console.error(error));
+  }
+
+  function userAdd() {
+    invoke('auth_user_add')
+      .then(() => {
+        fetchUser();
+      })
+      .catch((error) => console.error(error));
+  }
+
+  function userRemove() {
+    invoke('auth_remove_user', { user: user.id })
+      .then((user = undefined))
+      .catch((error) => console.error(error));
+  }
+
+  onMount(() => {
+    fetchUser();
+  });
+</script>
+
+<div class="settings-accounts">
+  {#if user !== undefined}
+    {#if user.uninitialized}
+      <div>
+        <p>Loading accounts...</p>
+      </div>
+    {:else}
+      <div>
+        Username: {user.username}
+        <br />
+        UUID: {user.id}
+      </div>
+    {/if}
+  {:else}
+    <div>
+      <p>No accounts found.</p>
+    </div>
+  {/if}
+
+  <div class="button-row">
+    <Button
+      color="primary"
+      class="button-row__items"
+      disabled={user !== undefined}
+      on:click={userAdd}>Add an account</Button
+    >
+    <Button
+      color="primary"
+      class="button-row__items"
+      disabled={user === undefined || user.uninitialized}
+      on:click={userRemove}>Log out</Button
+    >
+  </div>
+</div>
+
+<style lang="postcss">
+  .settings-accounts {
```

Comment:
are inline styles for this necessary?

Reply:
It's just a demo/debugging interface, loosely based on existing pages and Omorphia. Can this be improved? Yes, but this wasn't the original purpose behind this PR.